### ISSUE TITLE
Feature request: map with array of methods

### ISSUE THREAD

**Issue by ccerrillo** [thephpleague/route#303]
I think it would be useful, allow using an array of methods, as it could be done before in [3.x version](https://github.com/thephpleague/route/issues/57)

**Comment by zkwbbr** [comment]
I like this feature too. Most of the time I have the same route for `GET` and `POST`. E.g.,

```
$router->get('register', Register::class);
$router->post('register', Register::class);
```

Would be nice if there's something like `$router->map(['GET', 'POST'], 'register', Register::class);`

**Comment by kimjohans** [comment]
+1

**Comment by kimjohans** [comment]
Because nobody seems to work on this project, i'll publish a temporary solution:

[https://www.yourhelpcenter.de/2021/12/php-league-router-um-post-get-erweitern/](https://www.yourhelpcenter.de/2021/12/php-league-router-um-post-get-erweitern/)

With modifying two files in the package, a new Method "->any()" is setting "->get()" and "->post()" in one step.

**Comment by philipobenito** [comment]
@kimjohans erm, pull requests are a thing my friend. It's not a feature that's high on my priority list, but if it's very important to you, it's very simple to send a PR.

**Comment by kimjohans** [comment]
@philipobenito ehm, my friend ? The last update on this project is nearly half a year ago, most times are shown in years and my request #323 has net even been answered in any form. In #323, i wrote "... (urgently) require nested routing groups, because ...", but that info had no effect.

I gave a (temporary) solution to this problem - saying "thanks" could be an option, too  ... my friend.
It's always a pleasure to help.

**Comment by philipobenito** [comment]
@kimjohans okay so first, your comment came across as a rude "nobody works on this" and a link to your blog post, to me, that's just advertising if there was no attempt at a PR or a specific question about the status.

Second, had you asked, I could have told you that this is a slow moving project, rarely gets immediate updates unless there is a PR with good justification that isn't overly complex, conveniences like this are looked at every few months when I have time to give them the attention they require and there is usually a major update around once a year to coincide with language/dependency updates.

Third, no convenience update is considered "urgent", since you mentioned your nested groups request, the previous issue talking about them is mentioned in that issue, so you already know my misgivings on it, something you specifically require isn't necessarily required by an open source project, you are free to submit a PR, a discussion will take place and if it makes sense it will be merged, but if not, then you are absolutely free to make use of your own fork.

Fourth, if your comment wasn't meant to be rude then I do apologise but it seriously comes off that way, just keep in mind that people work on open source software for free, they usually have full time jobs, their own personal issues, etc, so if you require a feature but don't attempt contributing to the project, at least be nice when asking, not "other frameworks do X (of course)", "nobody works on this", etc, that just comes across as "this is wrong", which is never going to get immediate attention. 

I'll say as I say to everyone, this project isn't a framework, it's designed to be built on top of, I welcome all contributions and will try to make them work as long as they make sense and don't introduce too much complexity, even if a PR is rejected at first attempt, if I figure out a way to implement it in a way that works in the future then I will. 

However, I don't think your comment and link to your blog is a contribution, if you've gone to the effort of making a change, then wrote about it, then I don't understand why not a PR, that's why it comes across as advertising.

Tomorrow I'll be deleting all the comments unrelated to the actual feature request, including your original comment, if you would like to contribute then please submit your change as a PR.

**Comment by ccerrillo** [comment]
Calm down guys, @philipobenito you are doing a good job for free, maybe @kimjohans it's not comfortable doing a PR, I can do it when I have some time

**Comment by philipobenito** [comment]
@ccerrillo I admit, caught me in a bad mood, most days wouldn't react, even sarcastically, there's just a way to approach OSS issues.

**Comment by philipobenito** [comment]
Off topic stuff deleted.

For clarity, I'm not a fan of this feature, separation of concerns is always gonna be better than handling multiple types of HTTP request in one controller.

I'm not completely against it being implemented though, but it is very low priority for me.

If someone wants to create a PR, I would suggest a simple proxy method

`any(array $methods, $controller)`

and it should loop the methods array and invoke `map`.

**Comment by kimjohans** [comment]
That's new: I posted a solution/workaround to this request with link to a website with a detailed solution (nearly a PR) and the answer is deleted because it does not make the admin happy !

One honest question: What is the goal of this project ?
Helping users with improvements, they (!) need or keeping a version which cannot be used in bigger project - or will limit projects growing ?

Of course, you may find this question unpolite or whatever. But developing software FOR users or customers is always something, you have do deal with user interests and critics. Simply deleting unwelcome comments and solutions may work in some cultures/countries/groups ... but it always will keep people away from helping you to improve your project and use your work in other projects. If i would handle our customers critics like you did with my solution, i would better look for another job.

**Comment by philipobenito** [comment]
@kimjohans comments were deleted as they were unrelated to the issue. You caught me in a bad mood with a rude comment and a link to your blog, I won't rehash what I commented to you last night, but I included your original comment in the deletions as I don't think it actually helps.

The goal of this project... is that I built something for me, and released it for others to use, if they wish to, nobody has to, nobody is paying me to maintain the project, and I'm not motivated by solving absolutely everyone's specific use cases, this is not a framework. 

No user of open source software is entitled to the specific feature they want, equating it to customers is nonsense, you're not my customer, you're someone who has installed an open source package, and whether intended or not, your comments on issues come across as entitled, using language like "this is required", "will this just be closed like the other?", "X is wrong", "since nobody is working on this", normally, this sort of stuff would just be ignored, but it has escalated from what could just be translation issues, to something I found rude, and honestly felt like you just advertising your blog.

Even now, you're suggesting that it cannot be used in a larger project, neither of the issues we're discussing prevent use in large projects, they just prevent you doing something in the way that YOU want to do it. That isn't how open source works, there are reasons not to implement these features, but as mentioned, I'm not against them if a solution I'm happy with can be provided, I won't be making them a priority for me though, if they are for you... PR, and we can go through a review process.

TL;DR
1. Just because something isn't the way you like it, doesn't mean it needs to change.
2. If you'd like a change, contribute and provide a workable solution that takes everything in to account. (A link to a workaround isn't this).
3. You are not a customer of an open source maintainer, they are not required to work for you, nor to accept rude comments.
4. If I overreacted to the initial comment, although it was only a little sarcasm, then I apologise, but you've kind of proven my point since that.

Again, because this is off topic, I will be deleting it later.

**Comment by kimjohans** [comment]
Read not your whole comment, because the mistakes starts in your 2nd sentence:
It is not my blog, it is a german help website. i just found a solution there and linked it.
i wrote comments there to inform the authors there ... 

But truth is something difficult .. it is easier to delete comments and critics quickly.

**Comment by philipobenito** [comment]
@kimjohans I'm not gonna argue with you. I think you should find a different router to use.

If you had read the whole comment, you'd see that I apologised for my sarcasm in my initial response, but you really have justified my thoughts that lead to that sarcasm.

No open source maintainer owes you a single thing, especially when you haven't contributed, that includes listening to you piss and moan or allowing issues to go off topic with that pissing and moaning, that's not deleting criticism, it's deleting noise.
